### ISSUE TITLE
batch_run module double multiplies iteration when running

### ISSUE THREAD

**Issue by clumdee** [projectmesa/mesa#1309]
**Describe the bug**
Module `batch_run` module encounters `iterations` twice.

**Expected behavior**
If `iterations=1` gets `2it`, then `iterations=2` gets `4it`, and `iterations=3` gets `6it`. 

**To Reproduce**
Simplified version of [tutorial](https://mesa.readthedocs.io/en/latest/tutorials/intro_tutorial.html#batch-run).
Set `iterations=1` gets `2it`, set `iterations=2` gets `8it`, set `iterations=3` gets `18it`. 

```python
params = {"width": 10, "height": 10, "N": [1, 2]}

results = batch_run(
    MoneyModel,
    parameters=params,
    iterations=1,
    max_steps=1,
    number_processes=1,
    data_collection_period=-1,
    display_progress=True,
)
```

**Additional context**
`iterations` here should be removed
https://github.com/projectmesa/mesa/blob/006359b2a7e9a3b550b790c1b16c5a089521bb27/mesa/batchrunner.py#L70

**Comment by rht** [comment]
Did you meant this line? https://github.com/projectmesa/mesa/blob/006359b2a7e9a3b550b790c1b16c5a089521bb27/mesa/batchrunner.py#L78=

**Comment by rht** [comment]
Yeah, appreciate a PR.
It needs to be `iterations` removed in L78 instead, because the multi-process version of the loop in L98-110 uses `kwargs_list` without the nested loop of `for iteration in range(iteration)` and `for kwargs in kwargs_list`.

**Comment by clumdee** [comment]
I see. Then we need to find a way to properly loop in L85-95. Because leaving L70 as is will keep the single-process loop to continue double counting `iterations`.

I can't even get the multi-process to run on my machine. Not sure if it is related to this issue.

**Comment by rht** [comment]
You can leave the multiprocess version intact, and simply remove the `for iteration in range(iterations)` in the single process version, because the `kwargs_list` variable has already been multiplied by `iterations`. If you are on Windows, you need to do this extra step: https://github.com/projectmesa/mesa/blob/main/docs/useful-snippets/snippets.rst#using-multi-process-batch_run-on-windows= (this is also mentioned in the tutorial in case you missed it)

**Comment by clumdee** [comment]
Thanks for the suggestion. 
However removing `for iteration in range(iterations)` will affect L91. 

Hence, we need to find another way to track iteration logging. 
Maybe change L86 to enum?

I am on a mac so I didn't try that. But will give it a shot.

**Comment by rht** [comment]
> However removing for iteration in range(iterations) will affect L91.

That's true. Ugh. Ideally, we want the code for the single-process and multi-process version to be as similar as possible. Might have to use `Counter()` as well.

> I am on a mac so I didn't try that. But will give it a shot.

Gotcha. This is not yet documented in that snippets.rst, but for macOS, you need to put your main running code inside of `if __name__ == "__main__"`. See https://github.com/projectmesa/mesa/discussions/1202#discussioncomment-2567669=

**Comment by rht** [comment]
Just to emphasize: no need to do the `freeze_support()` thing.

**Comment by clumdee** [comment]
I have made a PR. 
Thanks for the tip on running multi-process.

**Comment by rht** [comment]
Closed via #1310.
